code/modules/reagents/reagent_containers/food/drinks.dm

```diff
@@ -299,6 +299,31 @@ If you add a drink with an empty icon sprite, ensure it is in the same folder, e
 	pickup_sound = 'sound/items/pickup/papercup.ogg'
 	possible_transfer_amounts = null
 	volume = 30
+	/**
+	 * Details written on the cup, a la IRL coffee places
+	 */
+	var/list/details = list("Customer" = null, "Order" = null)
+
+/obj/item/reagent_containers/food/drinks/takeaway_cup_idris/attackby(obj/item/attacking_item, mob/user)
+	if(attacking_item.ispen())
+		var/choice = tgui_input_list(user, "Which detail to edit?", "Detail edit", list("Customer", "Order"))
+		switch(choice)
+			if("Customer")
+				details["Customer"] = tgui_input_text(user, "Enter customer name", "Enter name")
+			if("Order")
+				details["Order"] = tgui_input_text(user, "Enter ordered drink", "Enter order")
+		name = initial(name)
+		if(details["Order"])
+			name += " - [details["Order"]]"
+		if(details["Customer"])
+			name += " ([details["Customer"]])"
```

Comment:
this should probably not be in the name, because it would lead to some huge names:
`takeaway cup - caffé macchiato - Winds of the Flowing Natural River`